pkg/wwan/mmagent/mmdbus/client.go

```diff
@@ -1252,42 +1265,63 @@ func (c *Client) Connect(
 	if err == nil {
 		return ipSettings, nil
 	}
-	origErr := err
-	// Try to fix failing connection attempt.
-	// First check if modem can even register.
-	changed, err := c.reconfigureEpsBearerIfNotRegistered(modemObj, connProps)
-	if changed && err == nil {
-		// Retry connection attempt with the same parameters applied also for the initial
-		// EPS bearer.
-		ipSettings, err = c.runSimpleConnect(modemObj, connProps)
-		if err == nil {
-			return ipSettings, nil
-		}
+	origErr := fmt.Errorf("failed to connect modem %s to APN %s: %v",
+		modemPath, args.APN, err)
+	// TODO: Allow the user to configure IP-type and PDP context for the initial EPS bearer.
+	// For now, we will be retrying connection attempts using all three IP types:
+	// ipv4, ipv6, ipv4v6. However, we will not be modifying the PDP context parameters
+	// for the initial EPS bearer until user is able to configure them.
+	//
+	// Some background for understanding: LTE connection consists of two IP bearers,
+	// the initial EPS bearer and the default EPS bearer. Device must first
+	// establish the initial bearer (which shows as transition from the "searching" to
+	// "registered" state) and then it connects to a default bearer (transition from
+	// "registered" to "connected"). Both bearers require PDP context settings (APN,
+	// ip-type, potentially username/password, etc.).
+	// Settings for the initial bearer are typically provided by the SIM card while
+	// settings for the default bearer are user-configured.
+	//
+	// It is not necessarily the case that the APNs for the initial and default bearers
+	// are the same. We used to make that assumption but this has led to cases where modem
+	// was failing registration because the APN for the initial bearer was wrong. It is
+	// better to let the SIM card provide the PDP context setting for the initial EPS bearer.
+	// Furthermore, once these settings are changed, there is no straightforward method to
+	// revert back to the SIM-provided configuration; for more details, see the discussion here:
+	// https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/1490#note_2628804
+	//
+	// Users may only need to override SIM-provided settings in rather rare cases: either when
+	// the SIM card has incorrect configuration (we have seen this only once) or when the initial
+	// EPS bearer requires username/password authentication (also uncommon). Despite the rarity
```

Comment:
When the EPS bearer requires authentication, will the username/password be the same for the initial and default bearers? Or can there we cases where we will need two sets of username/password?

Reply:
In theory, the credentials can be different (although I haven't seen this in practice yet).
Windows already allows to set different APN/IP-type/username/password for these two bearers (they call them "Attach" and "Internet"). It is in "Cellular Settings" under "Advanced Options". Soon Ubuntu/GNOME will support this as well.

Reply:
OK, so when we add this we should put in an additional set of credentials in the API.

Reply:
Yes, correct.